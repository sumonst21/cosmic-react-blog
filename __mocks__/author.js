export default {
  _id: '5afdbf6c44d74b4e924f1d09',
  bucket: '5afdbcd115bef94f7421eb0d',
  slug: 'john-doe',
  title: 'John Doe',
  content: '<p>Something about John...</p>',
  metafields: [
    {
      value: '9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg',
      key: 'image',
      title: 'Image',
      type: 'file',
      children: null,
      url: 'https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg',
      imgix_url: 'https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg',
    },
  ],
  type_slug: 'authors',
  created: '2017-10-12T13:27:49.665Z',
  created_at: '2017-10-12T13:27:49.665Z',
  modified_at: '2018-05-17T17:44:12.977Z',
  status: 'published',
  published_at: '2018-05-17T17:44:12.977Z',
  metadata: {
    image: {
      url: 'https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg',
      imgix_url: 'https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg',
    },
  },
};
