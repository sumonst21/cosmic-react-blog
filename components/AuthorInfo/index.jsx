import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

const styles = theme => ({
  content: {
    [`@media screen and (max-width: ${theme.breakpoints.values.sm}px)`]: {
      flexDirection: 'column',
      marginLeft: 0,
      marginTop: theme.spacing.unit * 2,
    },
    marginLeft: theme.spacing.unit * 4,
    marginTop: 0,
  },
  image: {
    boxShadow: '4px 4px 0px 0px rgba(0,0,0,0.7)',
    display: 'block',
    flex: 'none',
    height: 120,
    margin: 0,
    padding: 0,
    width: 120,
  },
  root: {
    [`@media screen and (max-width: ${theme.breakpoints.values.sm}px)`]: {
      flexDirection: 'column',
    },
    alignItems: 'center',
    display: 'flex',
    margin: '0 auto',
  },
});

const AuthorInfo = ({
  classes,
  className,
  title,
  content,
  metadata,
}) => {
  const imageUrl = metadata && metadata.image && metadata.image.imgix_url;

  return (
    <div className={classnames([classes.root, className])}>
      {
        imageUrl &&
        <img
          alt={title}
          className={classes.image}
          height={120}
          src={imageUrl}
          width={120}
        />
      }
      {
        content &&
        <Typography
          classes={{ root: classes.content }}
          component="div"
          variant="body2"
        >
          {/* eslint-disable-next-line react/no-danger */}
          <span dangerouslySetInnerHTML={{ __html: content }} />
        </Typography>
      }
    </div>
  );
};

AuthorInfo.defaultProps = {
  className: undefined,
  content: undefined,
  metadata: undefined,
};

AuthorInfo.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  content: PropTypes.string,
  metadata: PropTypes.shape({
    image: PropTypes.shape({
      imgix_url: PropTypes.string,
    }),
  }),
};

export default withStyles(styles)(AuthorInfo);
