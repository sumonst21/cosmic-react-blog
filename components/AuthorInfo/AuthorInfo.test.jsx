import author from '../../__mocks__/author';
import AuthorInfo from '.';

describe('AuthorInfo', () => {
  it('should render', () => {
    const component = mount((
      <AuthorInfo
        {...author}
        className="customClassName"
      />
    ));
    return expect(toJson(component)).toMatchSnapshot();
  });
});
