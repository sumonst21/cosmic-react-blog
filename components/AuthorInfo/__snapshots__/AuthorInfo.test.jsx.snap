// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorInfo should render 1`] = `
<WithStyles(AuthorInfo)
  _id="5afdbf6c44d74b4e924f1d09"
  bucket="5afdbcd115bef94f7421eb0d"
  className="customClassName"
  content="<p>Something about John...</p>"
  created="2017-10-12T13:27:49.665Z"
  created_at="2017-10-12T13:27:49.665Z"
  metadata={
    Object {
      "image": Object {
        "imgix_url": "https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
        "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
      },
    }
  }
  metafields={
    Array [
      Object {
        "children": null,
        "imgix_url": "https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
        "key": "image",
        "title": "Image",
        "type": "file",
        "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
        "value": "9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
      },
    ]
  }
  modified_at="2018-05-17T17:44:12.977Z"
  published_at="2018-05-17T17:44:12.977Z"
  slug="john-doe"
  status="published"
  title="John Doe"
  type_slug="authors"
>
  <AuthorInfo
    _id="5afdbf6c44d74b4e924f1d09"
    bucket="5afdbcd115bef94f7421eb0d"
    className="customClassName"
    classes={
      Object {
        "content": "AuthorInfo-content-1",
        "image": "AuthorInfo-image-2",
        "root": "AuthorInfo-root-3",
      }
    }
    content="<p>Something about John...</p>"
    created="2017-10-12T13:27:49.665Z"
    created_at="2017-10-12T13:27:49.665Z"
    metadata={
      Object {
        "image": Object {
          "imgix_url": "https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
          "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
        },
      }
    }
    metafields={
      Array [
        Object {
          "children": null,
          "imgix_url": "https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
          "key": "image",
          "title": "Image",
          "type": "file",
          "url": "https://s3-us-west-2.amazonaws.com/cosmicjs/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
          "value": "9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg",
        },
      ]
    }
    modified_at="2018-05-17T17:44:12.977Z"
    published_at="2018-05-17T17:44:12.977Z"
    slug="john-doe"
    status="published"
    title="John Doe"
    type_slug="authors"
  >
    <div
      className="AuthorInfo-root-3 customClassName"
    >
      <img
        alt="John Doe"
        className="AuthorInfo-image-2"
        height={120}
        src="https://cosmic-s3.imgix.net/9a0a85b0-23f5-11e7-875c-3f5dc9c15c2b-male.jpg"
        width={120}
      />
      <WithStyles(Typography)
        classes={
          Object {
            "root": "AuthorInfo-content-1",
          }
        }
        component="div"
        variant="body2"
      >
        <Typography
          align="inherit"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter-17",
              "alignJustify": "MuiTypography-alignJustify-19",
              "alignLeft": "MuiTypography-alignLeft-16",
              "alignRight": "MuiTypography-alignRight-18",
              "body1": "MuiTypography-body1-13",
              "body2": "MuiTypography-body2-12",
              "button": "MuiTypography-button-15",
              "caption": "MuiTypography-caption-14",
              "colorError": "MuiTypography-colorError-27",
              "colorInherit": "MuiTypography-colorInherit-23",
              "colorPrimary": "MuiTypography-colorPrimary-24",
              "colorSecondary": "MuiTypography-colorSecondary-25",
              "colorTextSecondary": "MuiTypography-colorTextSecondary-26",
              "display1": "MuiTypography-display1-8",
              "display2": "MuiTypography-display2-7",
              "display3": "MuiTypography-display3-6",
              "display4": "MuiTypography-display4-5",
              "gutterBottom": "MuiTypography-gutterBottom-21",
              "headline": "MuiTypography-headline-9",
              "noWrap": "MuiTypography-noWrap-20",
              "paragraph": "MuiTypography-paragraph-22",
              "root": "MuiTypography-root-4 AuthorInfo-content-1",
              "subheading": "MuiTypography-subheading-11",
              "title": "MuiTypography-title-10",
            }
          }
          color="default"
          component="div"
          gutterBottom={false}
          headlineMapping={
            Object {
              "body1": "p",
              "body2": "aside",
              "display1": "h1",
              "display2": "h1",
              "display3": "h1",
              "display4": "h1",
              "headline": "h1",
              "subheading": "h3",
              "title": "h2",
            }
          }
          noWrap={false}
          paragraph={false}
          variant="body2"
        >
          <div
            className="MuiTypography-root-4 AuthorInfo-content-1 MuiTypography-body2-12"
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Something about John...</p>",
                }
              }
            />
          </div>
        </Typography>
      </WithStyles(Typography)>
    </div>
  </AuthorInfo>
</WithStyles(AuthorInfo)>
`;
